### PR TITLE
[browser][MT] install rejection handler on workers

### PR DESCRIPTION
**Pull request by pavelsavara**
- `installUnhandledErrorHandler` on WebWorker when `withExitOnUnhandledError()`
 - replace `worker.onerror` with `mono_exit` on UI thread when `withExitOnUnhandledError()`
 - improve `stringify_as_error_with_stack` to contain `reason.message`
 - improve `reason` to always have at least some stack, stringified
 - don't `appendElementOnExit` on WebWorker
 - wrap worker init methods in try-catch-abort

Contributes to https://github.com/dotnet/runtime/issues/76957

### CODE REVIEW

File: src/mono/browser/runtime/startup.ts [hunk 1]
```diff
@@ -168,31 +169,43 @@ function preInit(userPreInit: (() => void)[]) {
 }
 
 async function preInitWorkerAsync() {
-    mono_log_debug("worker initializing essential C exports and APIs");
+    if (!MonoWasmThreads) return;
```
Comment (maraf) [inline src/mono/browser/runtime/startup.ts:172]: How can this happen?

**Review by maraf — APPROVED** [review]
Looks good to me 👍

File: src/mono/browser/runtime/startup.ts (hunk 1 above)
Comment (pavelsavara) [reply src/mono/browser/runtime/startup.ts:172]: this is just to trim, tree-shake the code. I don't think it would be called.

**Comment by pavelsavara** [comment]
There is `WebWorkerTest.JSSynchronizationContext_Send_Post_Items_Cancellation` hanging [Log](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-97752-merge-2e80c52139bb4dfa97/WasmTestOnBrowser-System.Runtime.InteropServices.JavaScript.Tests/1/console.d6a0afb1.log?helixlogtype=result)

And `memory access out of bounds` [Log](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-97752-merge-632f7533e0d44c1baa/WasmTestOnBrowser-System.Threading.Tasks.Tests/1/console.dfd35ff1.log?helixlogtype=result)

Both known and not caused here.

**Pull request merged**